rpc/Cargo.toml

```diff
@@ -9,6 +9,7 @@ thiserror = "1.0.40"
 jsonrpsee = { version = "0.20", features = ["client-core", "macros"] }
 celestia-types = { workspace = true }
 serde = { version = "1.0.188", features = ["derive"] }
+libp2p = { version = "0.52.3", optional = true }
```

Comment:
I think this must be moved to `dev-dependencies`

Reply:
Moved :+1: 
Note that we cannot have optional dev-dependencies, so it'll always be pulled for tests (which I guess is a good thing, unless we want to feature gate tests too?)